controllers/grid/form/SwordDepositPointForm.inc.php

```diff
@@ -46,6 +46,7 @@ public function __construct(SwordPlugin $swordPlugin, $contextId, $depositPointI
 		$this->addCheck(new FormValidatorLocale($this, 'name', 'required', 'plugins.generic.sword.depositPoints.required.field'));
 		$this->addCheck(new FormValidator($this, 'swordUsername', 'required', 'plugins.generic.sword.depositPoints.required.field'));
 		$this->addCheck(new FormValidator($this, 'swordPassword', 'required', 'plugins.generic.sword.depositPoints.required.field'));
+		$this->addCheck(new FormValidator($this, 'swordApikey', 'required', 'plugins.generic.sword.depositPoints.required.field'));
```

Comment:
I think this is supposed to be optional, correct? You might need to use a `FormValidatorCustom` if there's some logic here, e.g. (username and password) OR (API key).

Reply:
Ok - that seems to me like the right logic to me. I've added 

1) a custom function on the server:
https://github.com/quoideneuf/sword/blob/api-token/controllers/grid/form/SwordDepositPointForm.inc.php#L47

2) a custom JS FormHandler:
https://github.com/quoideneuf/sword/blob/api-token/js/SwordEditDepositPointsFormHandler.js
The result should be a dynamic error message depending on the various ways the 3 fields can be invalid.